7_react/react1/week3/hw-react-week3/src/components/TaskList.jsx

```diff
@@ -0,0 +1,133 @@
+import { useState, useEffect } from "react";
+import Checkbox from "./Checkbox";
+import { DateTime } from "luxon";
+import EditForm from "./EditForm";
+
+const TaskList = (props) => {
+  const { list, setList } = props;
+  const [selectedTasks, setSelectedTasks] = useState([]);
+  const [editingTask, setEditingTask] = useState(null);
+  const [editingTaskId, setEditingTaskId] = useState(null);
+
+  useEffect(() => {
+    const fetchDefaultTasks = async () => {
+      try {
+        const response = await fetch(
+          "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+        );
+
+        if (!response.ok) {
+          throw new Error("Failed to fetch default tasks");
+        }
+
+        const data = await response.json();
+
+        const tasksWithIds = data.map((task, index) => ({
+          ...task,
+          id: String(index + 1),
+        }));
+        setList(tasksWithIds);
+      } catch (error) {
+        console.error("Error fetching default tasks:", error.message);
+      }
+    };
+
+    fetchDefaultTasks();
+  }, []);
+
+  const onChangeStatus = (element) => {
+    const { name, checked } = element.target;
+
+    const taskToUpdate = list.find((task) => task.id === name);
+
+    if (taskToUpdate) {
+      const updatedSelectedTasks = checked
+        ? [...selectedTasks, taskToUpdate]
+        : selectedTasks.filter((task) => task.id !== name);
+
+      setSelectedTasks(updatedSelectedTasks);
+
+      if (updatedSelectedTasks.length === 0) {
+        setEditingTaskId(null);
+      }
+
+      const updateList = list.map((item) => ({
+        ...item,
+        done: item.id === name ? checked : item.done,
+      }));
+      setList(updateList);
+    }
+  };
+
+  const onClickEditItem = (task) => {
+    setEditingTaskId(task.id === "" ? null : task.id);
```

Comment:
Have you accounted for `task.id` being undefined or null?
In general, for such checks, it is a good idea to use the Falsy concept in JavaScript (see https://developer.mozilla.org/en-US/docs/Glossary/Falsy).
Notice that this is very specific to JavaScript.